stdlib/public/SwiftShims/SwiftStddef.h

```diff
@@ -29,11 +29,19 @@ typedef __SIZE_TYPE__ __swift_size_t;
 #endif
 
 // This selects the signed equivalent of the unsigned type chosen for size_t.
+#if defined(_MSC_VER)
+#if defined(_M_X86) || defined(_M_ARM)
+typedef int ssize_t;
+#elif defined(_M_AMD64) || defined(_M_ARM64)
```

Comment:
`_WIN64` then `_WIN32`, maybe? Although I'm wondering why MSVC is ever even looking at this header; it should always be compiled with the just-built Clang, no?

Reply:
Whoah... its a wild @jrose-apple!

Reply:
@jrose-apple - nope, because this is getting indirectly included into the *compiler*.   It is really sad that we cannot limit this header to just the stdlib/runtime which I would really prefer.

Reply:
That seems…wrong. Maybe *that's* the thing worth tackling: figuring out which runtime / shims headers need to go into the compiler and separating them out.

Reply:
@jrose-apple I agree.  I had tried to remove the use of the header, but @jckarter didn't want to lose the re-use of the types defined in the headers IIRC.

Reply:
I don't think we would remove *all* of them, just split up the parts that need to be included by the compiler. Maybe those still use __swift_ssize_t, though.

Yet Another Option: `#if !defined(__swift__) /* just use the platform ssize_t */ #endif`.

Reply:
That's what I would recommend; the only place we need to avoid using the platform headers is when importing the header to build the standard library.

Reply:
`ssize_t` is not available on all platforms as it is not mandated by the standard, it is an extension.

Reply:
It's *probably* the same as `ptrdiff_t`, if SSIZE_MAX isn't defined?

Reply:
`SSIZE_MAX` is also not available IIRC.  Yes, we should be able to use `std::make_signed<std::size-t>::type` but obviously only in the C++ context.

Reply:
Yeah, I meant to use the absence of `SSIZE_MAX` as a signifier to use a fallback type, but in practice I think it'd be valid to say that these headers must be included in either a Swift or a C++ context. (Note that Swift should be tested first, since someday we'll have C++-in-Swift. Somehow.)